asset/asset.go

```diff
@@ -953,21 +954,49 @@ type GroupKeyReveal interface {
 // that includes the specified data. If the data is nil, the leaf will not
 // contain any data but will still be a valid non-spendable script leaf.
 //
-// The script leaf is made non-spendable by including an OP_RETURN opcode at the
-// start of the script. While the script can still be executed, it will always
-// fail and cannot be used to spend funds.
-func NewNonSpendableScriptLeaf(data []byte) (txscript.TapLeaf, error) {
-	// Construct a script builder and add the OP_RETURN opcode to the start
-	// of the script to ensure that the script is non-executable.
-	scriptBuilder := txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN)
+// The script leaf is made non-spendable by including an OP_RETURN/OP_CHECKSIG
+// opcode at the start of the script. While the script can still be executed, it
+// will always fail and cannot be used to spend funds.
+func NewNonSpendableScriptLeaf(version GkrV1Version,
+	data []byte) (txscript.TapLeaf, error) {
+
+	var builder *txscript.ScriptBuilder
+	switch version {
+	// For the OP_RETURN based version, we'll use a single OP_RETURN opcode.
+	case OpReturnVersion:
+		builder = txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN)
+		if data != nil {
+			builder = builder.AddData(data)
+		}
+
+	// For the Pedersen commitment based version, we'll use a single
+	// OP_CEHCKSIG with an un-spendable key.
+	case PedersenVersion:
+		var msg [sha256.Size]byte
+		copy(msg[:], data)
+
+		// Make a Pedersen opening that uses no mask (we don't carry on
+		// the random value, as we don't care about hiding here). We'll
+		// also use the existing NUMs point.
+		op := pedersen.Opening{
+			Msg: msg,
+		}
+		commitPoint := pedersen.NewCommitment(op).Point()
```

Comment:
I think this sets `r = 1` here. :+1: 

Which IMO is simpler than setting it to asset ID (which an earlier revision might have done?)

Reply:
So since we don't specify the option of the masking/blinding value, `r` gets set to one during the option handling in `NewCommitment`.